go/cmd/dolt/commands/diff.go

```diff
@@ -480,30 +507,48 @@ func printShowCreateTableDiff(ctx context.Context, td diff.TableDelta) errhand.V
 	}
 
 	if fromCreateStmt != toCreateStmt {
-		cli.Println(textdiff.LineDiff(fromCreateStmt, toCreateStmt))
+		if fromCreateStmt == "" && toSch.GetAllCols().Size() > fromSch.GetAllCols().Size() {
+			if filter == FilterInserts || filter == NoFilter {
+				cli.Println(textdiff.LineDiff(fromCreateStmt, toCreateStmt))
+			}
+		} else if toCreateStmt == "" && toSch.GetAllCols().Size() < fromSch.GetAllCols().Size() {
+			if filter == FilterDeletes || filter == NoFilter {
+				cli.Println(textdiff.LineDiff(fromCreateStmt, toCreateStmt))
+			}
+		} else if filter == FilterUpdates || filter == NoFilter {
+			cli.Println(textdiff.LineDiff(fromCreateStmt, toCreateStmt))
+		}
 	}
 
 	return nil
 }
 
 // TODO: this doesn't handle check constraints or triggers
-func sqlSchemaDiff(ctx context.Context, td diff.TableDelta, toSchemas map[string]schema.Schema) errhand.VerboseError {
+func sqlSchemaDiff(ctx context.Context, td diff.TableDelta, toSchemas map[string]schema.Schema, filter string) errhand.VerboseError {
 	fromSch, toSch, err := td.GetSchemas(ctx)
 	if err != nil {
 		return errhand.BuildDError("cannot retrieve schema for table %s", td.ToName).AddCause(err).Build()
 	}
 
 	if td.IsDrop() {
-		cli.Println(sqlfmt.DropTableStmt(td.FromName))
+		if filter == NoFilter || filter == FilterDeletes {
```

Comment:
(minor) If you go with the `diffTypeFilter` struct idea above, this could be simple enough to roll into the existing if statement above; something like `if td.IsDrop() && diffTypeFilter.includeDrops() {`